### ISSUE TITLE
Armor Effects Staying On Player when Armor is Removed

### ISSUE THREAD

**Issue by Boolyman** [TheIllusiveC4/ConstructsArmory#20]
We are doing some beta testing for my upcoming pack, and noticed that the Iron Armor leaves the player with a magnetized effect even when taking the armor off.  This is problematic when players are trying to discard items... they go as far as to take the armor off, and still suck the items back up.  Can you do something to make sure the buff from the armor leaves when the armor is taken off?  Even better, I would recommend adding a function to the magnetize effect, that the player doesnt pick up what the just dropped for a few seconds.

Here is the Twtich clip of the issue:
https://clips.twitch.tv/FantasticVictoriousLasagnaTheThing?tt_medium=create

**Comment by Boolyman** [comment]
Update: It seems that logging off and logging back on gets rid of the buff.  But it should be stuck on them at all.

**Comment by TheIllusiveC4** [comment]
Hm, I definitely tested for this in earlier builds with no problems. It looks like I'll have to revisit this. I'll do some testing tomorrow and get back to you on this.

**Comment by TheIllusiveC4** [comment]
So I just tested this. Unfortunately, I cannot replicate this bug. Removing the armor removes the buffs, just as expected. There are some preventative checks that I could code in to possibly solve your problem, but I'd like to at least identify what's going on first.

Can you list your Forge, Construct's Armory, and Tinker's Construct versions? Also, can you replicate this in an instance with only those mods? My first thought is something from another mod is causing weird compatibility issues.

As a side note, I'll add in that functionality to prevent players from picking up dropped items for a few seconds.

**Comment by TheIllusiveC4** [comment]
I just pushed out a new update. While I did not receive the necessary information to figure out how this issue came about, I figured I may as well implement some preventative measures. When a player changes their armor, all of the armor abilities will be recalculated from scratch to ensure that players only have abilities for armor they're currently wearing. Let me know if your issue still persists.
